packages/docs/src/components/MedplumCodeBlock.tsx

```diff
@@ -0,0 +1,103 @@
+import CodeBlock, { Props } from '@theme/CodeBlock';
+import * as React from 'react';
+
+const BLOCK_START_PATTERN = new RegExp('^//\\s*start-block\\s+([A-Za-z_-]*)', '');
+// const BLOCK_START_PATTERN = /^\/\/\s*start-block\s+([A-Za-z_-]*)/;
+const BLOCK_END_PATTERN = new RegExp('^//\\s*end-block\\s+([A-Za-z_-]*)', '');
+
+interface MedplumCodeBlockProps extends Props {
+  selectLines?: number[][];
+  selectBlocks?: string[] | string;
+}
+
+export default function MedplumCodeBlock({
+  children,
+  selectLines,
+  selectBlocks,
+  ...props
+}: MedplumCodeBlockProps): JSX.Element {
+  let code = children as string;
+
+  if (selectLines && selectBlocks) {
+    throw new Error("Only one of 'selectLines' or 'selectBlocks' can be specified");
+  }
+
+  let codeLines: (string | null)[] = code.split('\n');
+
+  // Get the line ranges for all the blocks that have been selected.
+  const linesToRemove = new Set<number>();
+  const blockRanges = extractBlocks(codeLines);
+
+  // Save the line numbers of the magic comments, so that they can be removed later
+  Object.values(blockRanges)
+    .flat()
+    .forEach((lineNumber) => linesToRemove.add(lineNumber));
+  if (selectBlocks) {
+    if (typeof selectBlocks === 'string') {
+      selectBlocks = selectBlocks.split(',').map((s) => s.trim());
+    }
+    selectLines = selectBlocks.map((block) => {
+      if (!(block in blockRanges)) {
+        throw new Error(`No such code block: '${block}'`);
+      }
+      return blockRanges[block];
+    });
+  }
+
+  // Set all the magic comment lines to 'null' to be filtered out later
+  // We do this to preserve line numbers before slicing codeLines
+  linesToRemove.forEach((lineNumber) => (codeLines[lineNumber - 1] = null));
+
+  // For each line range to be selected, grab the slice from the original code
+  if (selectLines) {
+    codeLines = selectLines.flatMap((range) =>
+      range.length === 1 ? codeLines[range[0] - 1] : codeLines.slice(range[0] - 1, range[1])
+    );
+  }
+
+  // Filter out lines that were set to null
+  codeLines = codeLines.filter((line) => line !== null);
+
+  code = codeLines.join('\n');
+  return <CodeBlock {...props}>{code}</CodeBlock>;
+}
+
+/**
+ * Returns an map from the block name to the start and stop lines for that block
+ * @param codeLines
+ */
+function extractBlocks(codeLines: string[]): Record<string, [number, number]> {
```

Comment:
I'm not sure if it's a huge deal, but this means that we'll re-parse the entire input file for each code block?

Reply:
Yeah, it would. I don't have a great solution around that right now. I'm not really sure how to share information between CodeBlocks